lib/api/BackbeatAPI.js

```diff
@@ -864,6 +864,7 @@ class BackbeatAPI {
                                 return done();
                             });
                     }
+                    entries.push({ key, member, StorageClass });
```

Comment:
I don't know what the "applyRetryLockKey` protects against, if just to avoid duplicates in the retry list, should be ok, just want to make sure you have considered what are the implications of not locking for this particular case.

Reply:
I am re-using this method to queue stalled entries which don't have any redis keys set.

Reply:
I actually went ahead and made the change, it's better to use the lock to avoid race.